docs/development/abi.md

```diff
@@ -0,0 +1,219 @@
+# Pyodide Platform ABI
+
+## ABIs
+
+### General
+
+Shared libraries must be linked with `-sSIDE_MODULE=1` or
```

Comment:
Would it be better to phrase this in terms of "All Python module symbols that form part of a binary module's interface must be exported; the Emscripten linker can be instructed to do this by passing `-sSIDE_MODULE=1` to export *all* symbols, or ..."

i.e., capturing the high level requirement, rather than the specific interpretation?

Reply:
I'd also suggest that each high-level requirement should be a separate heading, so that from a compliance point of view, it's clear that there are N requirements must be adhered to.